src/child.js

```diff
@@ -0,0 +1,45 @@
+var childProcess = require('child_process');
+var fs = require('fs');
+
+var stdoutFile = process.argv[2];
+var stderrFile = process.argv[3];
+var cmd = process.argv[4];
+var argString = process.argv[5];
+var optString = process.argv[6];
+var pipe = process.argv[7];
+
+var opts = JSON.parse(optString);
+var args = JSON.parse(argString);
+
+var c = childProcess.spawn(cmd, args, opts);
+
+// Wait to exit until all other work has been finished, like closing IO streams
+function delayedExit(code) {
+  setTimeout(function () {
+    process.exit(code); // let streams close before ending
```

Comment:
This is better, thanks!

On second thought, I'm not completely sure if `delayedExit` would've worked. It posts a timeout to work around this, but that might not actually work if streams are buffered after the process finishes.